master/internal/api_workspace.go

```diff
@@ -369,6 +373,22 @@ func (a *apiServer) PatchWorkspace(
 	insertColumns := []string{}
 	updatedWorkspace := model.Workspace{}
 
+	if req.Workspace.Name != nil {
+		workspaceName := req.Workspace.Name.Value
+		if len(workspaceName) < 1 {
+			return nil, status.Errorf(codes.InvalidArgument,
+				"name '%s' must be at least 1 character long", workspaceName)
+		}
+		if len(workspaceName) > 80 {
+			return nil, status.Errorf(codes.InvalidArgument,
+				"name '%s' must be at most 80 character long", workspaceName)
+		}
+		if len(strings.Trim(workspaceName, " ")) == 0 {
+			return nil, status.Error(codes.InvalidArgument,
+				"name must contain at least non-whitespace letter")
+		}
+	}
```

Comment:
should i make a workspace name validation function for `PostWorkspace` and `PatchWorkspace` or is it ok to have dup code? not sure how backend team handle it.

Reply:
@keita-determined  A workspace validation name function using a switch would be ideal! 

For example

```
func validateWorkspaceName(n *wrapperspb.StringValue) error {
	switch {
	case n == nil:
		return nil
	case len(n.Value) < 1:
		return status.Errorf(codes.InvalidArgument,
			"name '%s' must be at least 1 character long", n)
	default:
		return nil
	}
}
```
(add in the extra cases)

and then you can check and return the error in each Workspace function like so:

```
	if err := validateWorkspaceName(req.Workspace.Name); err != nil {
		return nil, err
	}
```

Reply:
>is it ok to have dup code?
don't duplicate code unless you have a good reason for that.